### DISCUSSION TITLE
how can bundle worker code into a single js file?

### DISCUSSION THREAD

**Discussion by cbbfcd** [vikejs/vike#875] [Discussion]
i want to run server code in a edge node , and how can i bundle worker code into a single js file? i use tencent cloud

**Comment by brillout** [comment]
I recommend [esbuild](https://esbuild.github.io/), e.g. https://github.com/brillout/build-worker.

**Comment by cbbfcd** [comment]
Thanks for the reply, now I have another problem, how to deploy it?  that the bundle `dist` dir contains `client`, `server`, now add a `worker.js`,  how do I deploy code in `server`?

**Comment by cbbfcd** [comment]
maybe i should bundle all the code in `server` into `worker.js`?

**Comment by cbbfcd** [comment]
Correction, should I put build/server, build/client in the CDN, and then somehow configure build/worker.js to access them, because I have to use a single file worker, and I'm using a different service than cloudflare

**Reply by brillout** [reply]
It really depends on your deployment provider/strategy.

In general:
 - `dist/client/` should be statically served.
 - Running your server (e.g. `node ./server/my-express-app.js`) should just work (the files `/dist/server/**/*` are automatically imported thanks to https://github.com/brillout/vite-plugin-import-build).

FYI https://github.com/brillout/vite-plugin-ssr/discussions/789#discussioncomment-5907859.

**Comment by cbbfcd** [comment]
Thanks for the answer, I am facing a different scenario, I have a question, based on the cloudflare worker deployment demo, I understand that through the wranger deployment packaged a single worker bundle, and then uploaded dist/client to a static storage bucket like cos, but where did the dist/server go?

**Reply by brillout** [reply]
It's bundled by `wrangler`. Keep in mind that `dist/server/` is automagically included in the bundle thanks to https://github.com/brillout/vite-plugin-import-build.

**Comment by cbbfcd** [comment]
So I'm supposed to upload dist/client to the CDN, then configure baseAssets to point to this address, then deploy worker.js to the edge side rendering platform, and then I'm at a loss as to what to do with the build/worker dir, which the ESR platform requires to be a single bundle. And I don't have a tool like wranger yet

**Reply by brillout** [reply]
Sounds like you're on the right track.

As I said previously, if your deployment provider doesn't have something like `wrangler`, then implement something like https://github.com/brillout/build-worker.
